src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,161 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        Node(T value) {
+            this.value = value;
+            Node<T> prev;
+            Node<T> next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> addedNode = new Node<>(value);
+        if (head == null) {
+            head = tail = addedNode;
+        } else {
+            addLastNode(addedNode);
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index, size + 1);
+        Node<T> addedNode = new Node<>(value);
+        if (head == null) {
+            head = tail = addedNode;
+        } else if (index == 0) {
+            addFirstNode(addedNode);
+        } else if (index == size) {
+            addLastNode(addedNode);
+        } else {
+            addMiddleNode(index, addedNode);
+        }
```

Comment:
you can can call method add(T value) instead two conditionals 
and [Don’t complicate if-else construction. ](https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/linked-list.html#dont-complicate-if-else-construction-detailed-explanation)

Reply:
not fixed